src/corefx/System.Globalization.Native/pal_calendarData.c

```diff
@@ -385,20 +343,34 @@ bool EnumSymbols(const char* locale,
     for (int32_t i = startIndex; i < symbolCount; i++)
     {
         UErrorCode ignore = U_ZERO_ERROR;
-        int symbolLen = udat_getSymbols(pFormat, type, i, nullptr, 0, &ignore);
+        int symbolLen = udat_getSymbols(pFormat, type, i, NULL, 0, &ignore) + 1;
 
-        std::vector<UChar> symbolBuf(symbolLen + 1, '\0');
+        UChar* symbolBuf = calloc(symbolLen, sizeof(UChar));
```

Comment:
>UChar* symbolBuf = calloc(symbolLen, sizeof(UChar)); [](start = 8, length = 52)

can we optimize this, instead of allocating inside every time inside the loop, we can just allocate once outside (maybe with 100 UChar) and then check inside the loop if we need to resize this buffer if needed. in most cases I wouldn't expect we'll resize anyway

Reply:
Makes sense. Current approach is to provide 1:1 replacement for existing code and reap the performance benefit of reduced code size of C vs. C++. If we need additional performance tweaks, that can be done separately?